classes/DistrictTriggerHelper.cls

```diff
@@ -0,0 +1,151 @@
+/**
+	Description 	:   This is the Helper class  for District__c object to populate
+                        the  JSON as well as Count of districs and states.
+	Created BY 	    :   HANS RAJ JANGID
+	Created Date 	:   10-07-2022
+	Revision Logs 	:   V_1.0 - Created
+			            V_1.1 - Dev Name : HANS RAJ JANGID
+			                    Date     : 17-08-2022
+**/
+public class DistrictTriggerHelper {
+    public class Districts {
+        public String districtName{get; set;}
+    }
+    public class States {
+        public Integer Total_Number_Of_Districts{get; set;}
+        public string stateName{get; set;}   
+        public List<Districts> districts{get; set;}
+    }
+
+    public static void districtCountOnState(List<District__c> newDisList, List <District__c> oldDisList) {
+        List<integer> countDistrictCountry = new List<integer>();
+        List<integer> listcountStateDistrict = new List<integer>();
+        States wrapperOfState;
+        Districts wrapperOfDistrict;
+        //new sets for storing the State id
+        Set<id> storeStateId = new Set<id>();
+        Set<id> storeNewStateId = new Set<id>();
+        Set<id> storeOldStateId = new Set<id>();
+
+        //new sets for storing the Country id
+        Set<id> storeCountryId = new Set<id>();
+        
+        //new list for adding total disticts in state & country
+        List<State__c> totalDistrictsInState = new List<State__c>();
+        List<Country__c> totalDistrictsInCountry = new List<Country__c>();
+
+        List<State__c> addState = new List<State__c>();
+        List<Country__c> addCountry = new List<Country__c>();
+        //insert and upadte Case
+        if(newDisList != null){
+            if(newDisList != null ){
+                for (District__c district : newDisList){
+                    storeStateId.add(district.State_Name__c);
+                    storeCountryId.add(district.Country__c );
+                }
+            }
+            else if(newDisList != null && oldDisList!= null){
+                 for (District__c newDistrict : newDisList){
+                    for(District__c oldDistrict : oldDisList){
+        				if((newDistrict.State_Name__c != null) && (oldDisList == null || oldDistrict.State_Name__c != newDistrict.State_Name__c)) {                            
+                    		storeNewStateId.add(newDistrict.State_Name__c);
+                            storeOldStateId.add(oldDistrict.State_Name__c);		
+            			} 
+            		}
+        		} 
+                storeStateId.addAll(storeNewStateId);
```

Comment:
if in the end we need a single set then why are we separating and holding the id in two different sets?